src/chart/graph/GraphView.ts

```diff
@@ -131,21 +136,38 @@ class GraphView extends ChartView {
             const draggable = itemModel.get('draggable');
             if (draggable) {
                 el.on('drag', () => {
-                    if (forceLayout) {
-                        forceLayout.warmUp();
-                        !this._layouting
-                            && this._startForceLayoutIteration(forceLayout, layoutAnimation);
-                        forceLayout.setFixed(idx);
-                        // Write position back to layout
-                        data.setItemLayout(idx, [el.x, el.y]);
+                    switch (layout) {
+                        case 'force':
+                            forceLayout.warmUp();
+                            !this._layouting
+                                && this._startForceLayoutIteration(forceLayout, layoutAnimation);
+                            forceLayout.setFixed(idx);
+                            // Write position back to layout
+                            data.setItemLayout(idx, [el.x, el.y]);
+                            break;
+                        case 'circular':
+                            // mark node fixed
+                            node.getModel<GraphNodeItemOption>().mergeOption({fixed: true});
```

Comment:
Needs to reset the fixed value after `dragend`

Reply:
Now store in the layout object